doc/01_intro.md

```diff
@@ -0,0 +1,37 @@
+
+## An example
+
+An example may be useful to clarify.  In this example, we trace a single request to turn the volume up via the watch's music app.  From the user's point of view, here's a very simplified sequence diagram.  This assumes that the sofware is already running on the computer, that the watch is [paired](@ref paired), and that the watch is [connected](@ref connected).  The music player shown here isn't any particular software, but a class of software that can play media.  Real examples of such software include [vlc](https://www.videolan.org/) and [Rhythmbox](https://wiki.gnome.org/Apps/Rhythmbox), but as we will see later, as long as the software implements a particular kind of interface, it can be used.
+
+\startuml
+actor user
+user -> Music: press Volume Up
+Music -> Computer: "increase volume"
+Computer -> music_player:  increase volume
+\enduml
+
+From the developer's point of view, the story is a little more complex.  Since this is the `androidsyncservice` project, we will only describe the reception of a Bluetooth message and trace as this signal is processed on the computer.  First, the message is received by the computer's Bluetooth hardware.  A low level driver, which depends on the particular hardware used, takes care of this part and hands off the message to [BlueZ](http://www.bluez.org/) which is the official Linux Bluetooth protocol stack. 
+The BlueZ software then translates this incoming message into a [D-Bus](https://www.freedesktop.org/wiki/Software/dbus/) messages.  D-Bus is essentially a message manager that allows Linux-based software packages to talk to other Linux-based software packages.  In this instance, BlueZ handles many of the underlying complexity of Bluetooth for us.  The AsteroidOS uses [Bluetooth Low Energy](@ref ble) or BLE, to communicate.  BLE uses the General Attribute Profile (GATT) which collects individual data items called [characteristics](@ref characteristic) into [services](@ref service).  The `asteroidsyncserviced` software subscribes to notifications of changes for particular characteristics of interest.  When a characteristic changes, the software is notified.  In this particular example, that change is received by the MediaService::onCharacteristicChanged function. The notification also contains a passed value which is the desired volume level percentage expressed as an integer from 0 to 100.  This results in a signal, MediaService::volume being emitted, which is connected to the Platform::onMediaVolumeChange function.  That function then uses D-Bus once again to send a request to a media player to set its volume.  Unlike the MediaService class, however, the interface for the media player expects to receive the volume setting as a `double` in the range 0.0 to 1.0 so we divide by 100 to scale the value appropriately.
```

Comment:
_asteroidsyncserviced_ first occurrence of the service with a postfixed with the letter **d**.

Reply:
I changed it to *asteroidsyncservice* daemon for clarity.